practice-app/app/views.py

```diff
@@ -59,4 +60,41 @@ def WorldTime():
     else:
         return render_template("worldtime.html")
 
+def get_game_id_from_name(game_name):
+    steam_api_game_list_url = "https://api.steampowered.com/ISteamApps/GetAppList/v2/"
+    response = requests.get(steam_api_game_list_url).json()
+    all_games_list = response["applist"]["apps"]
+    game_name = "".join(game_name.lower().split())
+    for game in all_games_list:
+        if game_name == "".join(game["name"].lower().split()): 
+            return game["appid"]
+    return -1
 
+def get_game_information_from_id(game_id): 
+    steam_api_get_game_url = "https://store.steampowered.com/api/appdetails?appids={game_id}"
+    response = requests.get(steam_api_get_game_url.format(game_id=game_id)).json()
+    game_information = response[str(game_id)]["data"]
+    return game_information
+
+def get_game_information_from_name(game_name): 
+    game_id = get_game_id_from_name(game_name)
+    if game_id == -1: 
+        return None
+    game_information = get_game_information_from_id(game_id)
+    return game_information
+
+@app.route("/get_game_information", methods= ["GET", "POST"])
+def get_game_information():
+    error = None
+    if request.method == "POST":
+        game_name = request.form.get("game_name")
+        if not game_name:
+            error = 'Game name is required!'
+        else:
+            game_information = get_game_information_from_name(game_name)
+            if game_information == None: 
+                error = "Game is not found!"
+            else: 
+                # print(json.dumps(game_information, indent=4))
+                return render_template('show_game_information.html', response=game_information) 
+    return render_template("get_game_information.html", error=error)
```

Comment:
Looking at here, having a GET request automatically returns an error... is this intended? If yes, no problem.

Reply:
Yes, it is intended. For the GET request, the error returns as None and it is not shown in the html template. For the POST request, if there is an error it shows the error in the html.